sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -160,6 +160,16 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       }
     }
   }
+
+  // This is a test case for ORC-285
```

Comment:
This is the only issue for the new ORC reader, right? 

For the other two ORC reading modes, it works well, right? Could you also added test cases for the others too?

If this is the case, this should be a regression we have to merge to 2.3

Reply:
Yes. This is only issue for the new ORC reader, but I added this test case here for both `OrcSourceSuite` and `HiveOrcSourceSuite` to make it sure.

For Hive tables, if it needed, we had to create another test case. I'll try.

Reply:
yeah, for hive table, we might need a new test case

Reply:
After a bit of digging, the problem that ORC-285 fixes was introduced in HIVE-9711. HIVE-9711 is included in Hive 1.2.0, but doesn't create a problem unless the vectorized ORC reader is used. That didn't become the default behavior until HIVE-11417, which was released in Hive 2.1.0. So Spark's other readers are fine as long as they don't use the vectorized reader.

Reply:
Thank you, @omalley . Yes. It is. There is one difference in Spark.

Currently, Spark uses the vectorized reader for simple schema only. For `array<float>` or `array<double>` columns, we use ORC MR reader. However, MR reader uses vectorized RowBatch internally and it fails here.